### PR TITLE
Add fixed height option(Addresses #270)

### LINKED ISSUE #270: [Feature Request] fixed height
Now that we can spawn `bemenu` in the center of the screen, it would be useful to have an option that allows for spawning it with a fixed height when using `--list`.

Currently, I spawn `bemenu` in the middle of the screen, with 10 lines. As soon as I type something that yields less than 10 results, the whole menu will snap into a new position. This is visually distracting (and doesn't look too good, in my opinion).

### PR DESCRIPTION
**Pull request by stackform**
Add a option to make the menu have a fixed height while in vertical mode using `--fixed-height`. This prevents the menu from shrinking or expanding while filtering through items.

I modified how the filtered item count is set for the counter(getting the filtered item count directly rather than relying on menu->filtered.count) for the sake of efficiency, and it fixed a bug with the counter in vertical mode.

Closes #270

### CODE REVIEW

**Comment by dynamicdotvoid** [comment]
@Cloudef Any update on this?

File: lib/renderers/cairo_renderer.h [hunk 1]
```diff
@@ -297,7 +297,9 @@ bm_cairo_paint(struct cairo *cairo, uint32_t width, uint32_t max_height, const s
     uint32_t border_radius = menu->border_radius;
 
     uint32_t total_item_count = menu->items.count;
-    uint32_t filtered_item_count = (menu->filter ? menu->filtered.count : total_item_count);
+    //uint32_t filtered_item_count = (menu->filter ? menu->filtered.count : total_item_count);
+    uint32_t filtered_item_count;
+    bm_menu_get_filtered_items(menu, &filtered_item_count);
```
Comment (Cloudef) [inline lib/renderers/cairo_renderer.h:302]: why was this changed?

File: lib/renderers/cairo_renderer.h [hunk 2]
```diff
@@ -382,18 +392,24 @@ bm_cairo_paint(struct cairo *cairo, uint32_t width, uint32_t max_height, const s
...
-            } else if (i % 2 == 1) {
+            } else if (i < count && i % 2 == 1) {
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ALTERNATE_FG, &paint.fg);
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ALTERNATE_BG, &paint.bg);
             } else {
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ITEM_FG, &paint.fg);
                 bm_cairo_color_from_menu_color(menu, BM_COLOR_ITEM_BG, &paint.bg);
             }
+            
+            char *line_str = "error!";
```
Comment (Cloudef) [inline lib/renderers/cairo_renderer.h:406]: Don't quite understand why this is initialized as "error!", as the if / else below will always either set it to empty or to a formed message.

Comment (dynamicdotvoid) [reply lib/renderers/cairo_renderer.h:406]: I didn't really think of that, I just thought I should have it there, but looking at it it's always going to get initialized.

File: lib/renderers/cairo_renderer.h (hunk 1 above)
Comment (dynamicdotvoid) [reply lib/renderers/cairo_renderer.h:302]: I'm unsure about if it was a bug with the match counter or some code that I changed, but occasionally, the filtered items wouldn't update properly, so I decided to get the filtered item count from there and it seemed to fix the bug.

Comment (dynamicdotvoid) [reply lib/renderers/cairo_renderer.h:302]: On second thought, I'm pretty sure there was an issue with the ternary statement that incorrectly set the count to 0 occasionally. I believe this was an issue with the counter itself.

**Comment by dynamicdotvoid** [comment]
Should be ready to merge.

**Comment by stackform** [comment]
@Cloudef Any update on this? It's ready, and I've seen discussion about this feature on other issues.

**Comment by aarobc** [comment]
Please merge, this would be great to have.

**Comment by Cloudef** [comment]
Merged thanks!

**Comment by adastx** [comment]
--fixed-height option is only listed in the --help output, seem to be missing from the manpage.

**Comment by merrittlj** [comment]
I completely missed that. You can create a PR about it if you want, as I may not fix this anytime soon.

**Comment by adastx** [comment]
Created it :) #344.
